corehq/apps/userreports/models.py

```diff
@@ -760,7 +766,40 @@ def ui_filters(self):
     @property
     @memoized
     def charts(self):
-        return [ChartFactory.from_spec(g._obj) for g in self.configured_charts]
+        configured_charts = deepcopy(self.configured_charts)
+        for chart in configured_charts:
+            if chart['type'] == 'multibar':
```

Comment:
Can you say in a few words what the code in this if statement does?

And then perhaps, could it be extracted to a function named that thing?

Reply:
And even consider breaking it up further into other well-named functions? There is just a lot of data manipulation happening that is difficult to read/reason with.

Reply:
there are different types of charts: bar and pie. We are only catering for the case where the type is bar, referred to as multibar in code.